pkg/health/health_test.go

```diff
@@ -35,14 +35,19 @@ func getHealthStatus(yamlPath string, t *testing.T) *health.HealthStatus {
 	return health
 }
 
+func TestCrossplane(t *testing.T) {
+	assertAppHealth(t, "./testdata/crossplane.yaml", "Finished", health.HealthHealthy, true)
```

Comment:
This resource should have a warning health with status: ApplyFailure <message> - i.e. the LastAsyncOperation has higher priority 

The AsyncOperation condition is only for setting ready: false, it should never be used for status/health/message



```
  conditions:
    - type: Synced
      reason: ReconcileSuccess
      status: "True"
    - type: Ready
      reason: Available
      status: "True"
    - type: LastAsyncOperation
      reason: Success
      status: "True"
    - type: AsyncOperation
      reason: Finished
      status: "True"

```